lib/PuppeteerSharp/NetworkManager.cs

```diff
@@ -270,7 +270,7 @@ private void HandleRequestStart(string requestId, dynamic messageData)
 
         private void HandleRequestStart(string requestId, string interceptionId, string url, string resourceType, Payload requestPayload, string frameId)
         {
-            _frameManager.Frames.TryGetValue(frameId, out var frame);
+            _frameManager.Frames.TryGetValue(frameId ?? "", out var frame);
```

Comment:
Let's implement this here https://github.com/kblok/puppeteer-sharp/pull/110/files#diff-c363ddac9f50e1502fca9e72725d7090R273

So we avoid trying to get an item that we know we're not going to get.